### PR TITLE
Added signed and unsigned fixed point codecs for floating point scalars

### PR DESCRIPTION
**Pull request by jbrd**
Opening this PR to get some early feedback. Not ready to merge yet - need to add unit tests...

### CODE REVIEW

File: openvdb_points/openvdb.cc [hunk 1]
```diff
@@ -90,8 +90,10 @@ initialize()
 
     // fixed point compression
 
-    TypedAttributeArray<Vec3<float>, FixedPointCodec<true> >::registerType();
-    TypedAttributeArray<Vec3<float>, FixedPointCodec<false> >::registerType();
+    TypedAttributeArray<float, FixedPointCodec<true, UnitRange> >::registerType();
```
Comment (danrbailey) [inline openvdb_points/openvdb.cc]: I think we should register both 8-bit and 16-bit unit types for this.

File: openvdb_points_houdini/houdini/SOP_OpenVDB_Points.cc [hunk 2]
```diff
@@ -181,6 +182,10 @@ convertAttributeFromHoudini(PointDataTree& tree, const PointIndexTree& indexTree
                 convertAttributeFromHoudini<TypedAttributeArray<float,
                                             TruncateCodec>, false>(tree, indexTree, name, attribute, defaults);
             }
+            else if (compression == UNSIGNED_FIXED_POINT_8) {
```
Comment (danrbailey) [inline openvdb_points_houdini/houdini/SOP_OpenVDB_Points.cc]: Suggest we add support for both 8-bit and 16-bit to Houdini

File: openvdb_points_houdini/houdini/SOP_OpenVDB_Points.cc [hunk 3]
```diff
@@ -654,6 +663,7 @@ newSopOperator(OP_OperatorTable* table)
             "none", "None",
             "truncate", "16-bit Truncate",
             UnitVecCodec::name(), "Unit Vector",
+            FixedPointCodec<true, UnitRange>::name(), "8-bit Unit Fixed Point",
```
Comment (danrbailey) [inline openvdb_points_houdini/houdini/SOP_OpenVDB_Points.cc]: I would simply call this "8-bit Unit" and "16-bit Unit"

File: openvdb_points/openvdb.cc (hunk 1 above)
Comment (jbrd) [inline openvdb_points/openvdb.cc]: Done.

File: openvdb_points_houdini/houdini/SOP_OpenVDB_Points.cc (hunk 2 above)
Comment (jbrd) [inline openvdb_points_houdini/houdini/SOP_OpenVDB_Points.cc]: Done.

File: openvdb_points_houdini/houdini/SOP_OpenVDB_Points.cc (hunk 3 above)
Comment (jbrd) [inline openvdb_points_houdini/houdini/SOP_OpenVDB_Points.cc]: Done.

**Pull request merged**